### PR TITLE
refactor(translations): front-facing message translations

### PR DESCRIPTION
**Pull request by rumblefrog**
This moves a good majority of the front-facing messages to translations.

## Motivation and Context

Allow ease of translations of hard-coded values. Also closes #582 

## How Has This Been Tested?

This has not been tested so far.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: game/addons/sourcemod/scripting/sbpp_main.sp [hunk 1]
```diff
@@ -1187,7 +1188,7 @@ public void VerifyInsert(Database db, DBResultSet results, const char[] error, D
 	{
 		char length[32];
 		if(time == 0)
-			FormatEx(length, sizeof(length), "permament");
+			FormatEx(length, sizeof(length), "%t", "Permanent");
 		else
 			FormatEx(length, sizeof(length), "%d %s", time, time == 1 ? "minute" : "minutes");
```
Comment (TheByKotik) [inline game/addons/sourcemod/scripting/sbpp_main.sp]: Maybe need to remove hardcoded strings "minute" and "minutes"?

**Comment by CrazyHackGUT** [comment]
This is really good idea - move log messages to phrases and don't set obviously used language?
At this moment, your plugin can format log phrases on various languages if someone plugin used `SetGlobalTransTarget()`, and admin has translation to another languages.

**Review by Groruk — APPROVED** [review]

**Comment by rumblefrog** [comment]
@CrazyHackGUT Not sure if I understand the phrasing, but are you proposing that I do set global translation target.

**Comment by CrazyHackGUT** [comment]
I propose use `%T` instead `%t` and `LANG_SERVER` (SourceMod define; targets to default SM locale (en; can be changed in `core.cfg`)) as `client` for log entries, for example.

File: game/addons/sourcemod/scripting/sbpp_main.sp [hunk 2]
```diff
@@ -1187,9 +1188,10 @@ public void VerifyInsert(Database db, DBResultSet results, const char[] error, D
 	{
 		char length[32];
 		if(time == 0)
-			FormatEx(length, sizeof(length), "permament");
+			FormatEx(length, sizeof(length), "%t", "permanent");
```
Comment (TheByKotik) [inline game/addons/sourcemod/scripting/sbpp_main.sp]: Need use `FormatEx(length, sizeof(length), "%T", "Permanent", client);` for correct translation.

File: game/addons/sourcemod/scripting/sbpp_main.sp [hunk 3]
```diff
@@ -1187,9 +1188,10 @@ public void VerifyInsert(Database db, DBResultSet results, const char[] error, D
...
 		else
-			FormatEx(length, sizeof(length), "%d %s", time, time == 1 ? "minute" : "minutes");
+			FormatEx(length, sizeof(length), "%d %t", time, time == 1 ? "minute" : "minutes");
+
```
Comment (TheByKotik) [inline game/addons/sourcemod/scripting/sbpp_main.sp:1194]: `FormatEx(length, sizeof(length), "%d %T", time, time == 1 ? "minute" : "minutes", client);`

File: game/addons/sourcemod/scripting/sbpp_main.sp [hunk 4]
```diff
@@ -2625,40 +2619,48 @@ stock void PrepareBan(int client, int target, int time, char[] reason, int size)
...
 		}
-		LogAction(client, target, "\"%L\" banned \"%L\" (minutes \"%d\") (reason \"%s\")", client, target, time, reason);
+
+		LogAction(client, target, "%t", "Ban Log", client, target, time, reason);
+
 		char length[32];
+
 		if(time == 0)
-			FormatEx(length, sizeof(length), "permament");
+			FormatEx(length, sizeof(length), "%t", "permanent");
```
Comment (TheByKotik) [inline game/addons/sourcemod/scripting/sbpp_main.sp]: `FormatEx(length, sizeof(length), "%T", "Permanent", target);`

File: game/addons/sourcemod/scripting/sbpp_main.sp [hunk 5]
```diff
@@ -2625,40 +2619,48 @@ stock void PrepareBan(int client, int target, int time, char[] reason, int size)
...
 		else
-			FormatEx(length, sizeof(length), "%d %s", time, time == 1 ? "minute" : "minutes");
+			FormatEx(length, sizeof(length), "%d %t", time, time == 1 ? "minute" : "minutes");
```
Comment (TheByKotik) [inline game/addons/sourcemod/scripting/sbpp_main.sp]: `FormatEx(length, sizeof(length), "%d %T", time, time == 1 ? "minute" : "minutes", target);`

**Comment by TheByKotik** [comment]
> I propose use `%T` instead `%t` and `LANG_SERVER` (SourceMod define; targets to default SM locale (en; can be changed in `core.cfg`)) as `client` for log entries, for example.

No need that.
Both functions ( [LogToFile()](https://github.com/alliedmodders/sourcemod/blob/e02e6bcb4fa1f157b7bc32514ce203e5f5076720/core/logic/smn_core.cpp#L563) and [LogAction()](https://github.com/alliedmodders/sourcemod/blob/e02e6bcb4fa1f157b7bc32514ce203e5f5076720/core/logic/smn_core.cpp#L531) ) call this on C++ side.

**Pull request merged**